src/Microsoft.IdentityModel.Tokens/RsaSecurityKey.cs

```diff
@@ -27,69 +27,110 @@
 
 using System;
 using System.Security.Cryptography;
+using Microsoft.IdentityModel.Logging;
 
 namespace Microsoft.IdentityModel.Tokens
 {
     public class RsaSecurityKey : AsymmetricSecurityKey
     {
-        private RSAParameters _rsaParamaeters;
+        PrivateKeyStatus _privateKeyStatus = PrivateKeyStatus.AvailabilityNotDetermined;
```

Comment:
Since @brentschmaltz doesn't like the private enum, you could use a nullable boolean here :smile: